crates/runtime/src/imports.rs

```diff
@@ -1,46 +1,17 @@
 use crate::vmcontext::{VMFunctionImport, VMGlobalImport, VMMemoryImport, VMTableImport};
-use wasmtime_environ::entity::{BoxedSlice, PrimaryMap};
-use wasmtime_environ::wasm::{FuncIndex, GlobalIndex, MemoryIndex, TableIndex};
 
 /// Resolved import pointers.
-#[derive(Clone)]
-pub struct Imports {
+#[derive(Clone, Default)]
+pub struct Imports<'a> {
     /// Resolved addresses for imported functions.
-    pub functions: BoxedSlice<FuncIndex, VMFunctionImport>,
+    pub functions: &'a [VMFunctionImport],
```

Comment:
A subtle consequence of switching from `BoxedSlice`/`PrimaryMap`/etc. to plain `&[]` is that `BoxedSlice`/`PrimaryMap` enforce the invariant that they only be indexed with `FuncIndex` indices. It looks like we don't directly index into the slices here, so that seems ok, but could you add comments here mentioning what the index space for these slices is?